django/db/backends/oracle/operations.py

```diff
@@ -436,14 +436,17 @@ def adapt_timefield_value(self, value):
                                value.second, value.microsecond)
 
     def combine_expression(self, connector, sub_expressions):
-        "Oracle requires special cases for %% and & operators in query expressions"
+        lhs, rhs = sub_expressions
         if connector == '%%':
             return 'MOD(%s)' % ','.join(sub_expressions)
         elif connector == '&':
             return 'BITAND(%s)' % ','.join(sub_expressions)
         elif connector == '|':
-            lhs, rhs = sub_expressions
             return 'BITAND(-%(lhs)s-1,%(rhs)s)+%(lhs)s' % {'lhs': lhs, 'rhs': rhs}
+        elif connector == '<<':
+            return '(%(lhs)s * POWER(2, %(rhs)s))' % {'lhs': lhs, 'rhs': rhs}
+        elif connector == '>>':
+            return 'FLOOR(%(lhs)s / POWER(2, %(rhs)s))' % {'lhs': lhs, 'rhs': rhs}
```

Comment:
I think that I found complex solution for all binary operators in `MySQL` that return an unsigned 64-bit integer. We can simple convert return value to `SIGNED` integer. I don't know why, but it doesn't work for right shift operator.
```diff
--- a/django/db/backends/mysql/operations.py
+++ b/django/db/backends/mysql/operations.py
@@ -202,8 +202,14 @@ class DatabaseOperations(BaseDatabaseOperations):
+        lhs, rhs = sub_expressions
         if connector == '^':
             return 'POW(%s)' % ','.join(sub_expressions)
+        # MySQL's binary operators return an unsigned 64-bit integer.
+        elif connector in ['<<', '|', '&']:
+            return 'CONVERT(%s, SIGNED)' % connector.join(sub_expressions)
+        elif connector == '>>':
+            return 'FLOOR(%(lhs)s / POW(2, %(rhs)s))' % {'lhs': lhs, 'rhs': rhs}
         return super(DatabaseOperations, self).combine_expression(connector, sub_expressions)
```